Gems/Atom/Feature/Common/Code/Source/DisplayMapper/DisplayMapperPass.cpp

```diff
@@ -243,13 +243,10 @@ namespace AZ
             AZStd::string azshaderPath = shaderFilePath;
             AZ::StringFunc::Path::ReplaceExtension(azshaderPath, "azshader");
 
-            Data::AssetId shaderAssetId;
-            Data::AssetCatalogRequestBus::BroadcastResult(
-                shaderAssetId, &Data::AssetCatalogRequestBus::Events::GetAssetIdByPath,
-                azshaderPath.c_str(), azrtti_typeid<RPI::ShaderAsset>(), false);
+            Data::AssetId shaderAssetId = AZ::RPI::AssetUtils::GetAssetIdForProductPath(azshaderPath.c_str(), AZ::RPI::AssetUtils::TraceLevel::Assert, azrtti_typeid<RPI::ShaderAsset>());
```

Comment:
Author's note, GetAssetIdForProductPath causes a blocking escalation, while simply asking the catalog does not.